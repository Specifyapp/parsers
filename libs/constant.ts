import { TokensType } from '../types';

export const tokenTypesList: Array<TokensType> = [
  'border',
  'color',
  'gradient',
  'duration',
  'measurement',
  'opacity',
  'shadow',
  'textStyle',
  'depth',
];
