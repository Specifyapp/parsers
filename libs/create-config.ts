import { Parser, Token } from '../types';

type UnknownParser<F> = { options: unknown; fn: F; name: string };
type ReturnPipe<T> = {
  run: (tokens: Array<Token>) => T;
  build: () => unknown;
};

export function createConfig<A extends unknown[], B>(
  ab: UnknownParser<(...a: A) => B>,
): ReturnPipe<B>;
export function createConfig<A extends unknown[], B, C>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
): ReturnPipe<C>;
export function createConfig<A extends unknown[], B, C, D>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
): ReturnPipe<D>;
export function createConfig<A extends unknown[], B, C, D, E>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
): ReturnPipe<E>;
export function createConfig<A extends unknown[], B, C, D, E, F>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
): ReturnPipe<F>;
export function createConfig<A extends unknown[], B, C, D, E, F, G>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
): ReturnPipe<G>;
export function createConfig<A extends unknown[], B, C, D, E, F, G, H>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
  gh: UnknownParser<(g: G) => H>,
): ReturnPipe<H>;
export function createConfig<A extends unknown[], B, C, D, E, F, G, H, I>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
  gh: UnknownParser<(g: G) => H>,
  hi: UnknownParser<(h: H) => I>,
): ReturnPipe<I>;
export function createConfig<A extends unknown[], B, C, D, E, F, G, H, I, J>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
  gh: UnknownParser<(g: G) => H>,
  hi: UnknownParser<(h: H) => I>,
  ij: UnknownParser<(i: I) => J>,
): ReturnPipe<J>;

export function createConfig(...parsers: Array<UnknownParser<(...params: unknown[]) => unknown>>): {
  run: (tokens: Array<Token>) => unknown;
  build: () => unknown;
} {
  return {
    build: () =>
      parsers.map(parser => {
        const step: Parser = {
          name: parser.name,
        };
        if (parser.options) {
          step.options = parser.options;
        }
        return step;
      }),
    run: (tokens: Array<Token>): unknown => {
      return parsers.reduce<unknown>(async (value, parser) => parser.fn(await value), tokens);
    },
  };
}
