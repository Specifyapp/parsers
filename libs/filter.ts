import { AtLeast, TokensType, TokenTypeFromTokenTypeName } from '../types';

export function filter<T extends TokensType>(
  tokens: Array<AtLeast<{ type: TokensType }>>,
  type: T,
) {
  return tokens.filter(token => token.type === type) as Array<TokenTypeFromTokenTypeName<T>>;
}
