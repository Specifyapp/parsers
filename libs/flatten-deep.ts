export function flattenObject(obj: Record<string, any>) {
  const flattened: Record<string, any> = {};

  Object.keys(obj).forEach(key => {
    if (typeof obj[key] === 'object' && obj[key] !== null) {
      Object.assign(flattened, flattenObject(obj[key]));
    } else {
      flattened[key] = obj[key];
    }
  });

  return flattened;
}
