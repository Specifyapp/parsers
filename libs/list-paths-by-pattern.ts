import { has, get } from 'lodash';

export default function listPathsByPattern<T extends object>(
  base: T | Array<T>,
  pattern: string,
): Array<string> {
  try {
    if (!pattern.includes('[*]')) return has(base, pattern) ? [pattern] : [];
    return pattern
      .split(/(?=\[\*])|(?<=\[\*])/g) // split without remove the delimiter
      .reduce<Array<string>>(
        (paths, item) => {
          if (item === '[*]') {
            return paths
              .map(basePath => {
                const iteration = basePath ? get(base, basePath) : base;
                if (!Array.isArray(iteration)) return [];
                return Array(iteration.length)
                  .fill(0)
                  .reduce(
                    (acc, _, index) =>
                      has(base, `${basePath}[${index}]`) ? [...acc, `${basePath}[${index}]`] : acc,
                    [],
                  );
              })
              .flat(2);
          } else {
            return paths.reduce<Array<string>>(
              (acc, elm) => (has(base, `${elm}${item}`) ? [...acc, `${elm}${item}`] : acc),
              [],
            );
          }
        },
        [''],
      );
  } catch (err) {
    throw err;
  }
}
