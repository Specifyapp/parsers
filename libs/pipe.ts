type UnknownFunction = (...params: unknown[]) => unknown;

// Copied from https://github.com/gcanti/fp-ts/blob/1.15.0/src/function.ts#L209 with some
// modifications to make the first function allow multiple parameters.
export function pipe<A extends unknown[], B>(ab: (this: void, ...a: A) => B): (...args: A) => B;
export function pipe<A extends unknown[], B, C>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
): (...args: A) => C;
export function pipe<A extends unknown[], B, C, D>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
): (...args: A) => D;
export function pipe<A extends unknown[], B, C, D, E>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
): (...args: A) => E;
export function pipe<A extends unknown[], B, C, D, E, F>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
): (...args: A) => F;
export function pipe<A extends unknown[], B, C, D, E, F, G>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
): (...args: A) => G;
export function pipe<A extends unknown[], B, C, D, E, F, G, H>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
  gh: (this: void, g: G) => H,
): (...args: A) => H;
export function pipe<A extends unknown[], B, C, D, E, F, G, H, I>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
  gh: (this: void, g: G) => H,
  hi: (this: void, h: H) => I,
): (...args: A) => I;
export function pipe<A extends unknown[], B, C, D, E, F, G, H, I, J>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
  gh: (this: void, g: G) => H,
  hi: (this: void, h: H) => I,
  ij: (this: void, i: I) => J,
): (...args: A) => J;

export function pipe(...fns: UnknownFunction[]): UnknownFunction {
  return (...initialParams: unknown[]): unknown =>
    fns.reduce<unknown>((value, fn, index) => {
      const params = index === 0 ? (value as unknown[]) : [value];
      return fn(...params);
    }, initialParams);
}
