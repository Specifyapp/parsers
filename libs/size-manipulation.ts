import { MeasurementValue } from '../types';

function convertMeasurement(
  from: MeasurementValue,
  toUnit: 'px' | 'rem',
  basePixelValue?: number | undefined,
): MeasurementValue;
function convertMeasurement(
  from: string,
  toUnit: 'px' | 'rem',
  basePixelValue?: number | undefined,
): string;
function convertMeasurement(
  from: string | MeasurementValue,
  toUnit: 'px' | 'rem',
  basePixelValue = 16,
): MeasurementValue | string {
  if (!['px', 'rem'].includes(toUnit)) throw new Error('Unknown size unit');
  let fromValue = typeof from === 'object' && 'measure' in from ? from.measure : parseFloat(from);
  const fromUnit = typeof from === 'string' ? from.match(/[\d.\-\+]*\s*(.*)/)![1] || '' : from.unit;
  let pxValue;
  if (fromUnit === 'px') {
    pxValue = fromValue;
  } else if (fromUnit === 'rem') {
    pxValue = fromValue * basePixelValue;
  }
  if (!pxValue) throw new Error('Unknown size unit');

  const result = {
    unit: toUnit === 'px' ? 'px' : 'rem',
    measure: toUnit === 'px' ? pxValue : pxValue / basePixelValue,
  };
  return typeof from === 'string' ? `${result.measure}${result.unit}` : result;
}

export default convertMeasurement;
