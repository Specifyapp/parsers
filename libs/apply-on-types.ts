import { tokenTypesList } from './constant';
import { TokensType } from '../types';

export function getTokenTypesToApplyFn<T extends { filter?: { types?: Array<TokensType> } }>(
  options: T,
) {
  if (!options?.filter?.types) {
    return tokenTypesList;
  }
  return tokenTypesList.filter(tokenType => options.filter!.types!.includes(tokenType));
}
