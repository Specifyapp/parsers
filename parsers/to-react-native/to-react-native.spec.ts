import libs from '../global-libs';
import seeds from '../../tests/seeds';
import toReactNative, { InputDataType } from './to-react-native.parser';

describe('To React Native', () => {
  it('Should generate the theme object', async () => {
    const tokens = seeds().tokens;
    const result = await toReactNative(tokens, undefined, libs);
    expect(result).toMatchSnapshot();
  });
  it('Should use assetsFolderPath', async () => {
    const tokens = seeds().tokens;
    const result = await toReactNative(tokens, { assetsFolderPath: 'path' }, libs);
    expect(result).toMatchSnapshot();
  });
  it('Should support different duration types', async () => {
    const tokens: InputDataType = [
      {
        id: '1',
        name: '2 hours',
        value: { unit: 'h', duration: 2 },
        type: 'duration',
      },
      {
        id: '2',
        name: '2 minutes',
        value: { unit: 'm', duration: 2 },
        type: 'duration',
      },
      {
        id: '3',
        name: '2 seconds',
        value: { unit: 's', duration: 2 },
        type: 'duration',
      },
      {
        id: '4',
        name: '2 milliseconds',
        value: { unit: 'ms', duration: 2 },
        type: 'duration',
      },
      {
        id: '5',
        name: '2 nanoseconds',
        value: { unit: 'ns', duration: 2 },
        type: 'duration',
      },
    ];
    const result = await toReactNative(tokens, { assetsFolderPath: 'path' }, libs);
    expect(result).toMatchSnapshot();
  });
});
