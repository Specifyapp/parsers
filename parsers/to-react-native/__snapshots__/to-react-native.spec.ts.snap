// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To React Native Should generate the theme object 1`] = `
"const theme = {
  bitmap: {
    acmeLogo: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/13d6/2eb6/a2626d914998754ac0b704e2cdb7a813'\\",
    },
    photoExample: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/3302/0ec5/79870771c0861eaa8f1af4e4ed5a8790'\\",
    },
  },
  vector: {
    activity: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/96dc/8825/c166b559140a0a64b28441924700a0b2'\\",
    },
    airplay: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/ba89/d0aa/da2310c63c85699af43c8d128619cfdc'\\",
    },
    alertCircle: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/e8be/d9ce/4595ffa4510b7427d42e113c33632ae9'\\",
    },
    alertOctagon: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/fda9/f3ef/d31ed18fd324eb97190f34e0497214df'\\",
    },
    alertTriangle: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/880e/672c/155447c7e27185832ded7310ac9a02e7'\\",
    },
    alignCenter: {
      uri: \\"'https://s3-us-west-2.amazonaws.com/figma-alpha-api/img/44e1/3537/07081649495a48e2f2de80b83e69a876'\\",
    },
  },
  depth: {
    background: {
      elevation: 1,
      shadowOpacity: 0.1815,
      shadowRadius: 0.54,
      shadowOffset: { width: 0.6, height: 0.6 },
    },
    foreground: {
      elevation: 100,
      shadowOpacity: 0.32999999999999996,
      shadowRadius: 54,
      shadowOffset: { width: 60, height: 60 },
    },
    middle: {
      elevation: 50,
      shadowOpacity: 0.255,
      shadowRadius: 27,
      shadowOffset: { width: 30, height: 30 },
    },
  },
  duration: { base: 300, long: 700, short: 100, veryLong: 3000 },
  measurement: {
    baseSpace01: 4,
    baseSpace02: 8,
    baseSpace03: 12,
    baseSpace04: 16,
    baseSpace05: 32,
    baseSpace06: 48,
    baseSpace07: 64,
    baseSpace08: 80,
    baseSpace09: 96,
    baseSpace10: 112,
  },
  textStyle: {
    body: {
      fontWeight: \\"500\\",
      fontSize: 14,
      lineHeight: 20,
      fontFamily: \\"Inter-Medium\\",
      color: \\"rgb(30, 33, 43)\\",
      letterSpacing: 10,
    },
    bodyWithOpacity: {
      fontWeight: \\"500\\",
      fontSize: 14,
      lineHeight: 20,
      fontFamily: \\"Inter-Medium\\",
      color: \\"rgba(30, 33, 43, 0.3)\\",
      letterSpacing: 10,
    },
    code: {
      fontWeight: \\"500\\",
      fontSize: 13,
      lineHeight: 20,
      fontFamily: \\"FiraCode-Medium\\",
      color: \\"rgb(255, 142, 5)\\",
    },
    list: {
      fontWeight: \\"400\\",
      fontSize: 14,
      lineHeight: 20,
      fontFamily: \\"Roboto-Regular\\",
      color: \\"rgb(0, 0, 0)\\",
    },
    title: {
      fontWeight: \\"600\\",
      fontSize: 32,
      lineHeight: 40,
      fontFamily: \\"Inter-SemiBold\\",
      color: \\"rgb(30, 33, 43)\\",
    },
  },
  border: {
    borderAccent: {
      borderWidth: 2,
      borderStyle: \\"solid\\",
      borderColor: \\"rgb(102, 80, 239)\\",
      borderRadius: 16,
    },
    borderAccentWithOpacity: {
      borderWidth: 2,
      borderStyle: \\"solid\\",
      borderColor: \\"rgba(102, 80, 239, 0.5)\\",
      borderRadius: 16,
    },
    borderAccentWithoutRadii: {
      borderWidth: 2,
      borderStyle: \\"solid\\",
      borderColor: \\"rgba(102, 80, 239, 0.5)\\",
    },
    borderDashed: {
      borderWidth: 1,
      borderStyle: \\"dashed\\",
      borderColor: \\"rgb(30, 33, 43)\\",
      borderRadius: 16,
    },
  },
  color: {
    colorsAccent: \\"rgb(87, 124, 254)\\",
    colorsBlack: \\"rgb(30, 33, 43)\\",
    colorsGreen: \\"rgb(88, 205, 82)\\",
    colorsGrey: \\"rgb(204, 213, 225)\\",
    colorsOrange: \\"rgb(255, 142, 5)\\",
    colorsRed: \\"rgb(245, 72, 63)\\",
    colorsWhite: \\"rgb(255, 255, 255)\\",
  },
  font: {
    firaCodeMedium: \\"FiraCode-Medium\\",
    interMedium: \\"Inter-Medium\\",
    interSemiBold: \\"Inter-SemiBold\\",
    robotoRegular: \\"Roboto-Regular\\",
  },
  gradient: {
    gradientsColored: [
      {
        angle: 90,
        colors: [\\"rgb(245, 72, 63)\\", \\"rgb(255, 142, 5)\\"],
        locations: [0, 1],
      },
    ],
    gradientsDark: [
      {
        angle: 90,
        colors: [\\"rgb(30, 33, 43)\\", \\"rgb(204, 213, 225)\\"],
        locations: [0, 1],
      },
    ],
    gradientsNeutral: [
      {
        angle: 90,
        colors: [\\"rgb(204, 213, 225)\\", \\"rgb(255, 255, 255)\\"],
        locations: [0, 1],
      },
    ],
    gradientsSafari: [
      {
        angle: 90,
        colors: [\\"rgb(255, 142, 5)\\", \\"rgb(255, 255, 255)\\"],
        locations: [0, 1],
      },
    ],
  },
  opacity: { subtle: 0.5, transparent: 0.1, visible: 0.95 },
};
export default theme;
"
`;

exports[`To React Native Should support different duration types 1`] = `
"const theme = {
  duration: {
    \\"2Hours\\": 7200000,
    \\"2Minutes\\": 120000,
    \\"2Seconds\\": 2000,
    \\"2Milliseconds\\": 2,
    \\"2Nanoseconds\\": 0.002,
  },
};
export default theme;
"
`;

exports[`To React Native Should use assetsFolderPath 1`] = `
"import assetAcmeLogo from \\"path/acmeLogo.png\\";
import assetPhotoExample from \\"path/photoExample.png\\";
import assetActivity from \\"path/activity.svg\\";
import assetAirplay from \\"path/airplay.svg\\";
import assetAlertCircle from \\"path/alertCircle.svg\\";
import assetAlertOctagon from \\"path/alertOctagon.svg\\";
import assetAlertTriangle from \\"path/alertTriangle.svg\\";
import assetAlignCenter from \\"path/alignCenter.svg\\";

const theme = {
  bitmap: { acmeLogo: assetAcmeLogo, photoExample: assetPhotoExample },
  vector: {
    activity: assetActivity,
    airplay: assetAirplay,
    alertCircle: assetAlertCircle,
    alertOctagon: assetAlertOctagon,
    alertTriangle: assetAlertTriangle,
    alignCenter: assetAlignCenter,
  },
  depth: {
    background: {
      elevation: 1,
      shadowOpacity: 0.1815,
      shadowRadius: 0.54,
      shadowOffset: { width: 0.6, height: 0.6 },
    },
    foreground: {
      elevation: 100,
      shadowOpacity: 0.32999999999999996,
      shadowRadius: 54,
      shadowOffset: { width: 60, height: 60 },
    },
    middle: {
      elevation: 50,
      shadowOpacity: 0.255,
      shadowRadius: 27,
      shadowOffset: { width: 30, height: 30 },
    },
  },
  duration: { base: 300, long: 700, short: 100, veryLong: 3000 },
  measurement: {
    baseSpace01: 4,
    baseSpace02: 8,
    baseSpace03: 12,
    baseSpace04: 16,
    baseSpace05: 32,
    baseSpace06: 48,
    baseSpace07: 64,
    baseSpace08: 80,
    baseSpace09: 96,
    baseSpace10: 112,
  },
  textStyle: {
    body: {
      fontWeight: \\"500\\",
      fontSize: 14,
      lineHeight: 20,
      fontFamily: \\"Inter-Medium\\",
      color: \\"rgb(30, 33, 43)\\",
      letterSpacing: 10,
    },
    bodyWithOpacity: {
      fontWeight: \\"500\\",
      fontSize: 14,
      lineHeight: 20,
      fontFamily: \\"Inter-Medium\\",
      color: \\"rgba(30, 33, 43, 0.3)\\",
      letterSpacing: 10,
    },
    code: {
      fontWeight: \\"500\\",
      fontSize: 13,
      lineHeight: 20,
      fontFamily: \\"FiraCode-Medium\\",
      color: \\"rgb(255, 142, 5)\\",
    },
    list: {
      fontWeight: \\"400\\",
      fontSize: 14,
      lineHeight: 20,
      fontFamily: \\"Roboto-Regular\\",
      color: \\"rgb(0, 0, 0)\\",
    },
    title: {
      fontWeight: \\"600\\",
      fontSize: 32,
      lineHeight: 40,
      fontFamily: \\"Inter-SemiBold\\",
      color: \\"rgb(30, 33, 43)\\",
    },
  },
  border: {
    borderAccent: {
      borderWidth: 2,
      borderStyle: \\"solid\\",
      borderColor: \\"rgb(102, 80, 239)\\",
      borderRadius: 16,
    },
    borderAccentWithOpacity: {
      borderWidth: 2,
      borderStyle: \\"solid\\",
      borderColor: \\"rgba(102, 80, 239, 0.5)\\",
      borderRadius: 16,
    },
    borderAccentWithoutRadii: {
      borderWidth: 2,
      borderStyle: \\"solid\\",
      borderColor: \\"rgba(102, 80, 239, 0.5)\\",
    },
    borderDashed: {
      borderWidth: 1,
      borderStyle: \\"dashed\\",
      borderColor: \\"rgb(30, 33, 43)\\",
      borderRadius: 16,
    },
  },
  color: {
    colorsAccent: \\"rgb(87, 124, 254)\\",
    colorsBlack: \\"rgb(30, 33, 43)\\",
    colorsGreen: \\"rgb(88, 205, 82)\\",
    colorsGrey: \\"rgb(204, 213, 225)\\",
    colorsOrange: \\"rgb(255, 142, 5)\\",
    colorsRed: \\"rgb(245, 72, 63)\\",
    colorsWhite: \\"rgb(255, 255, 255)\\",
  },
  font: {
    firaCodeMedium: \\"FiraCode-Medium\\",
    interMedium: \\"Inter-Medium\\",
    interSemiBold: \\"Inter-SemiBold\\",
    robotoRegular: \\"Roboto-Regular\\",
  },
  gradient: {
    gradientsColored: [
      {
        angle: 90,
        colors: [\\"rgb(245, 72, 63)\\", \\"rgb(255, 142, 5)\\"],
        locations: [0, 1],
      },
    ],
    gradientsDark: [
      {
        angle: 90,
        colors: [\\"rgb(30, 33, 43)\\", \\"rgb(204, 213, 225)\\"],
        locations: [0, 1],
      },
    ],
    gradientsNeutral: [
      {
        angle: 90,
        colors: [\\"rgb(204, 213, 225)\\", \\"rgb(255, 255, 255)\\"],
        locations: [0, 1],
      },
    ],
    gradientsSafari: [
      {
        angle: 90,
        colors: [\\"rgb(255, 142, 5)\\", \\"rgb(255, 255, 255)\\"],
        locations: [0, 1],
      },
    ],
  },
  opacity: { subtle: 0.5, transparent: 0.1, visible: 0.95 },
};
export default theme;
"
`;
